depends/hosts/darwin.mk

```diff
@@ -1,4 +1,4 @@
-OSX_MIN_VERSION=10.14
+OSX_MIN_VERSION=10.15
```

Comment:
In commit "build: set OSX_MIN_VERSION to 10.15" (db29710bc12f97d9de78fb177322a8ac41f7c87d)

Would be good to consolidate all the config, makefile, and documentation changes adding build support for std::filesystem into a single atomic commit. Right now they are scattered across three commits and everything seems very mixed up:

- [ ] 7810d151f5219e634b431c5393771616c759aef7 refactor: replace boost::filesystem with std::filesystem (2/5)
- [ ] 9f37a39b11b240f75b97fb565e6d8cf44972ec49 build: remove boost::filesystem (3/5)
- [ ] db29710bc12f97d9de78fb177322a8ac41f7c87d build: set OSX_MIN_VERSION to 10.15 (4/5)

Better organization would be:

1. Add std::filesystem build support updating build files & minimum required versions
2. Switch c++ code from boost::filesystem to std::filesystem
3. Remove boost::filesystem